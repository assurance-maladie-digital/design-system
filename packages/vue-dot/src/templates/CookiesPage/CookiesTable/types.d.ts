export interface CookiesList {
	name: string;
	description: string;
	conservation: string;
}
