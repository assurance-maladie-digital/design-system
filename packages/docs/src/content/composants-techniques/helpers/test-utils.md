---
title: testUtils
description: Le helper `testUtils` contient un ensemble de fonctions utilitaires pour les tests unitaires.
---

<doc-tabs>

<doc-tab-item label="Utilisation">

<doc-alert type="info">

Le helper `testUtils` exporte également toutes les fonctions du package [Vue Test Utils](https://vue-test-utils.vuejs.org/).

</doc-alert>

### applyGlobalConfig

La fonction `applyGlobalConfig` permet d’appliquer la configuration globale pour les tests, elle est utilisée dans le fichier `setup.ts`.

```ts
import { applyGlobalConfig } from '@cnamts/vue-dot/src/helpers/testUtils';

applyGlobalConfig();
```

### installGlobalPlugins

La fonction `installGlobalPlugins` permet d’installer les plugins [Vue Input Facade](https://ronaldjerez.github.io/vue-input-facade/), [Vue Meta](https://vue-meta.nuxtjs.org/), [Vuex](https://vuex.vuejs.org/), [Vuetify](https://v2.vuetifyjs.com/) et Vue Dot pour les tests.

```ts
import { createLocalVue, installGlobalPlugins } from '@cnamts/vue-dot/src/helpers/testUtils';

const localVue = createLocalVue();

installGlobalPlugins(localVue);
```

### installRouter

La fonction `installRouter` permet d’installer [Vue Router](https://v3.router.vuejs.org/fr/) pour les tests.

```ts
import { createLocalVue, installRouter } from '@cnamts/vue-dot/src/helpers/testUtils';

const localVue = createLocalVue();

installRouter(localVue);
```

### createRouter

La fonction `createRouter` permet de créer une instance de [Vue Router](https://v3.router.vuejs.org/fr/) pour les tests.

```ts
import { createRouter } from '@cnamts/vue-dot/src/helpers/testUtils';

const routes = [];

const router = createRouter({
	routes
});
```

### createStore

La fonction `createStore` permet de créer une instance de [Vuex](https://vuex.vuejs.org/) pour les tests.

```ts
import { createStore } from '@cnamts/vue-dot/src/helpers/testUtils';

const store = createStore({
	state: {}
});
```

### createVuetifyInstance

La fonction `createVuetifyInstance` permet de créer une instance de [Vuetify](https://v2.vuetifyjs.com/) pour les tests.

```ts
import { createVuetifyInstance } from '@cnamts/vue-dot/src/helpers/testUtils';

const vuetify = createVuetifyInstance();
```

### mockVFormRef

La fonction `mockVFormRef` permet de mocker la référence d’un composant [VForm](https://v2.vuetifyjs.com/en/components/forms/) pour les tests.

```ts
import { mockVFormRef } from '@cnamts/vue-dot/src/helpers/testUtils';

const mocks = {
	$refs: {
		form: mockVFormRef(true)
	}
};
```

### mockTranslations

La fonction `mockTranslations` permet de mocker les traductions pour les tests.

```ts
import { mockTranslations } from '@cnamts/vue-dot/src/helpers/testUtils';

const mocks = {
	...mockTranslations<string | string[]>({
		'path.to.title': 'Titre',
		'path.to.items': [
			'Élément 1',
			'Élément 2',
			'Élément 3'
		]
	})
};
```

</doc-tab-item>

<doc-tab-item label="API">
<doc-api name="helpers/test-utils"></doc-api>
</doc-tab-item>

</doc-tabs>
