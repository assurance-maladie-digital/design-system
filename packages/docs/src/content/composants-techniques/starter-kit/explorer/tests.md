Le dossier `tests` contient les tests d’interface et les tests unitaires globaux de l’application.

<doc-alert type="warning">
Il n’est pas recommandé de renommer ce dossier.
</doc-alert>
