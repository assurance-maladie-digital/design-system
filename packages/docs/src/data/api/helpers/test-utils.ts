import { Api } from '~/types';

export const api: Api = {
	applyGlobalConfig: {
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	installGlobalPlugins: {
		arguments: [
			{
				name: 'localVue',
				description: 'Local Vue instance',
				type: 'Vue',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	installRouter: {
		arguments: [
			{
				name: 'localVue',
				description: 'Local Vue instance',
				type: 'Vue',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	createRouter: {
		arguments: [
			{
				name: 'options',
				description: 'Router options',
				type: 'RouterOptions',
				required: false,
			},
		],
		returnValue: [
			{
				type: 'VueRouter',
			},
		],
	},
	createStore: {
		arguments: [
			{
				name: 'options',
				description: 'Store options',
				type: 'StoreOptions',
				required: false,
			},
		],
		returnValue: [
			{
				type: 'Store',
			},
		],
	},
	createVuetifyInstance: {
		arguments: [
			{
				name: 'preset',
				description: 'Vuetify preset',
				type: 'Partial<UserVuetifyPreset>',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'Vuetify',
			},
		],
	},
	mockVFormRef: {
		arguments: [
			{
				name: 'isValid',
				description: 'Is valid',
				type: 'boolean',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	mockTranslations: {
		arguments: [
			{
				name: 'translations',
				description: 'Translations',
				type: 'IndexedObject',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
};
