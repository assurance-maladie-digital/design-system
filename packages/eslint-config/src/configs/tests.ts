import { type FlatESLintConfigItem } from 'eslint-define-config';
import { pluginVitest } from '../plugins';
import { GLOB_TESTS } from '../globs';
import { isInEditor } from '../env';

export const tests: FlatESLintConfigItem[] = [
	{
		plugins: {
			tests: pluginVitest,
		},
	},
	{
		files: GLOB_TESTS,
		rules: {
			'tests/consistent-test-it': [
				'error',
				{
					fn: 'it',
					withinDescribe: 'it',
				},
			],
			'tests/expect-expect': 'error',
			'tests/no-alias-methods': 'error',
			'tests/no-commented-out-tests': isInEditor ? 'off' : 'error',
			'tests/no-conditional-expect': 'error',
			'tests/no-disabled-tests': 'error',
			'tests/no-done-callback': 'error',
			'tests/no-duplicate-hooks': 'error',
			'tests/no-focused-tests': isInEditor ? 'off' : 'error',
			'tests/no-identical-title': 'error',
			'tests/no-interpolation-in-snapshots': 'error',
			'tests/no-standalone-expect': 'error',
			'tests/no-test-return-statement': 'error',
			'tests/prefer-comparison-matcher': 'error',
			'tests/prefer-equality-matcher': 'error',
			'tests/prefer-hooks-in-order': 'error',
			'tests/prefer-hooks-on-top': 'error',
			'tests/prefer-mock-promise-shorthand': 'error',
			'tests/prefer-spy-on': 'error',
			'tests/prefer-to-be-falsy': 'error',
			'tests/prefer-to-be-object': 'error',
			'tests/prefer-to-be-truthy': 'error',
			'tests/prefer-to-be': 'error',
			'tests/prefer-to-contain': 'error',
			'tests/prefer-to-have-length': 'error',
			'tests/prefer-todo': 'error',
			'tests/require-hook': 'error',
			'tests/require-top-level-describe': 'error',
			'tests/valid-describe-callback': 'error',
			'tests/valid-expect': 'error',
			'tests/valid-title': 'error',
		},
	},
];
