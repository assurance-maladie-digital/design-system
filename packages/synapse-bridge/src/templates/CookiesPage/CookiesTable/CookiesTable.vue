<script lang="ts">
	import { defineComponent } from 'vue'
	import type { PropType } from 'vue'

	import { headers } from './headers'

	import { CookiesList } from './types'

	export default defineComponent({
		props: {
			items: {
				type: Array as PropType<CookiesList[]>,
				required: true,
			},
		},
		data() {
			return {
				headers,
			}
		},
	})
</script>

<template>
	<VTable class="vd-cookie-table">
		<thead>
			<tr>
				<th
					v-for="(item, index) in headers"
					:key="index"
					:style="{ width: item.width }"
					class="text-left"
				>
					{{ item.label }}
				</th>
			</tr>
		</thead>

		<tbody>
			<tr
				v-for="(cookie, index) in items"
				:key="index"
			>
				<td>{{ cookie.name }}</td>
				<td>{{ cookie.description }}</td>
				<td>{{ cookie.conservation }}</td>
			</tr>
		</tbody>
	</VTable>
</template>

<style lang="scss" scoped>
	.vd-cookie-table :deep(table) {
		table-layout: fixed;
	}
</style>
