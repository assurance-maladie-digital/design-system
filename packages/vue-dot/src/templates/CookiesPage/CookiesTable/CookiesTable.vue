<template>
	<VSimpleTable class="vd-cookie-table">
		<thead>
			<tr>
				<th
					v-for="(item, index) in headers"
					:key="index"
					:style="{ width: item.width }"
					class="text-left"
				>
					{{ item.label }}
				</th>
			</tr>
		</thead>

		<tbody>
			<tr
				v-for="(cookie, index) in items"
				:key="index"
			>
				<td>{{ cookie.name }}</td>
				<td>{{ cookie.description }}</td>
				<td>{{ cookie.conservation }}</td>
			</tr>
		</tbody>
	</VSimpleTable>
</template>

<script lang="ts">
	import Vue, { PropType } from 'vue';
	import Component, { mixins } from 'vue-class-component';

	import { headers } from './headers';

	import { CookiesList } from './types';

	const Props = Vue.extend({
		props: {
			items: {
				type: Array as PropType<CookiesList[]>,
				required: true
			}
		}
	});

	const MixinsDeclaration = mixins(Props);

	@Component
	export default class CookiesPage extends MixinsDeclaration {
		headers = headers;
	}
</script>

<style lang="scss" scoped>
	.vd-cookie-table :deep(table) {
		table-layout: fixed;
	}
</style>
