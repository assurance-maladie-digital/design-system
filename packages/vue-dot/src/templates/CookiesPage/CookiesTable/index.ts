import CookiesTable from './CookiesTable.vue';

export default CookiesTable;
