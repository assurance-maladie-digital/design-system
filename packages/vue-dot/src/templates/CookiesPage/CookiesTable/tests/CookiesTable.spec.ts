import Vue from 'vue';
import { Wrapper } from '@vue/test-utils';

import { mountComponent } from '@/tests';

import CookiesTable from '../';

import { cookies } from '../../tests/data/cookiesList';

let wrapper: Wrapper<Vue>;

describe('CookiesTable', () => {
	it('renders correctly', () => {
		wrapper = mountComponent(CookiesTable, {
			propsData: {
				items: cookies
			}
		});

		expect(wrapper).toMatchSnapshot();
	});
});
