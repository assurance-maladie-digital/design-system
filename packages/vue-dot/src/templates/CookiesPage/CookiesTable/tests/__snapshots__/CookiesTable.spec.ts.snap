// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CookiesTable renders correctly 1`] = `
<vsimpletable-stub class="vd-cookie-table">
  <thead>
    <tr>
      <th class="text-left" style="width: 180px;">
        Dénomination du cookie
      </th>
      <th class="text-left" style="width: 328px;">
        Finalité
      </th>
      <th class="text-left" style="width: 180px;">
        Durée de conservation
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>lorem-ipsum</td>
      <td>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</td>
      <td>XX mois</td>
    </tr>
    <tr>
      <td>lorem-ipsum</td>
      <td>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</td>
      <td>XX mois</td>
    </tr>
  </tbody>
</vsimpletable-stub>
`;
